examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,692 @@
+        }
+    }
+}
+```
+
+Finally, register these listeners and start discovery.
+
+On Android, call `addCastingPlayerChangeListener` on the singleton instance of
+`MatterCastingPlayerDiscoverer` to register the listener, and then call
+`startDiscovery`.
+
+```java
+public static void demoDiscovery()
+{
+    // Get the singleton instance of the MatterCastingPlayerDiscoverer
+    CastingPlayerDiscoverer discoverer = MatterCastingPlayerDiscoverer.getInstance();
+
+    // Add our castingPlayerChangeListener to listen to changes in the discovered CastingPlayers
+    discoverer.addCastingPlayerChangeListener(castingPlayerChangeListener);
+
+    // Start discovery
+    discoverer.startDiscovery();
+
+    // After some time, stop discovering and remove our castingPlayerChangeListener
+    // from the set of listeners the CastingPlayerDiscoverer informs
+    executorService.schedule(() -> {
+        discoverer.stopDiscovery();
+        discoverer.removeCastingPlayerChangeListener(castingPlayerChangeListener);
+    },     30, TimeUnit.SECONDS);
+}
+```
+
+On iOS, register the listeners by calling `addObserver` on the
+`NotificationCenter` with the appropriate selector, and then call `start` on the
+`sharedInstance` of `MTRCastingPlayerDiscovery`.
+
+```objectivec
+func startDiscovery() {
+    NotificationCenter.default.addObserver(self, selector: #selector(self.didAddDiscoveredCastingPlayers), name: NSNotification.Name.didAddCastingPlayers, object: nil)
+    NotificationCenter.default.addObserver(self, selector: #selector(self.didRemoveDiscoveredCastingPlayers), name: NSNotification.Name.didRemoveCastingPlayers, object: nil)
+    NotificationCenter.default.addObserver(self, selector: #selector(self.didUpdateDiscoveredCastingPlayers), name: NSNotification.Name.didUpdateCastingPlayers, object: nil)
+
+    MTRCastingPlayerDiscovery.sharedInstance().start()
+    ...
+}
+```
+
+Note: You will need to connect with a Casting Player as described below to see
+the list of Endpoints that they support. Refer to the [Connection](#connection)
+section for details on how to discover available endpoints supported by a
+Casting Player.
+
+### Connect to a Casting Player
+
+_{Complete Connection examples:
+[Android](android/App/app/src/main/java/com/matter/casting/ConnectionExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRConnectionExampleViewModel.swift)}_
+
+Each CastingPlayer object created during [Discovery](#discovery) contains
+information such as deviceName, vendorId, productId, etc. which can help the
+user pick the right CastingPlayer. A Casting Client can attempt to connect to
+the selectedCastingPlayer using
+[Matter User Directed Commissioning (UDC)](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/rendezvous/UserDirectedCommissioning.adoc).
+The Matter TV Casting library locally caches information required to reconnect
+to a CastingPlayer, once it has been commissioned by it. After that, it is able
```

Comment:
...once the casting client has been commissioned by the CastingPlayer. After that, the casting client is able...

(eg. be clear what "it" was referring to)

Reply:
Done in project-chip#29870